(BKR-967) Add :disable_analytics option

james-stocks opened the pull request:
Allow :disable_analytics to be set in beaker options, allowing us to block
traffic to Google Analytics.

Help wanted:
- Is this the right way to merge PE specific options with beaker options?
- Should the pe_presets be inside a module instead of raw included code?

TODO:
- Still need to add spec tests

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

kevpl commented on lib/beaker-pe/install/pe_utils.rb:
```diff
@@ -702,9 +704,20 @@ def setup_defaults_and_config_helper_on(host, master, acceptable_exit_codes=nil)
         #Install PE based on global hosts with global options
         #@see #install_pe_on
         def install_pe
+          options.merge!(pe_presets)
```
this block should be at the top of `install_pe_on` rather than `install_pe`, otherwise people using the former won't get these changes. `install_pe` is really just a shortcut, it shouldn't have functionality in it.

kevpl commented on lib/beaker-pe/install/pe_utils.rb:
```diff
@@ -702,9 +704,20 @@ def setup_defaults_and_config_helper_on(host, master, acceptable_exit_codes=nil)
         def install_pe
+          options.merge!(pe_presets)
+
+          if options[:disable_analytics] == true then 
```
can just be
```ruby
if options[:disable_analytics]
```

kevpl commented on lib/beaker-pe/install/pe_utils.rb:
```diff
@@ -702,9 +704,20 @@ def setup_defaults_and_config_helper_on(host, master, acceptable_exit_codes=nil)
 
+        def disable_analytics(hosts)
+          logger.info("Disabling analytics on dashboard host(s)")
+          on(hosts, 'echo "127.0.0.1\tgoogle-analytics.com\n" >> /etc/hosts')
```
you should checkout beaker's `set_etc_hosts` method and see if that can do this for you, because it sets these host entries in a platform-independent manner.

kevpl commented:
@james-stocks, so in your testing of this change, you can set something in the CONFIG section of a hosts file, and it won't be overridden by your pe-presets? I just want to be 100% clear about that functionality before moving forward with this change.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/123/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/125/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/124/

james-stocks commented:
@kevpl I've updated according to your comments and have also updated the spec tests - can you also review those? I needed to expose the new pe_presets to the spec tests

In the meantime I will add some new spec tests and also test my changes. 
The expectation is that :disable_analytics will default to true for existing jobs that don't specify it. Test runs that require analytics can add the option to beaker options. I wasn't intending for the option to go into the host config; because it is a suite option and not a per-host option, but I will check that it still works when set in the hosts config instead of beaker options

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/126/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/127/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/128/

kevpl commented:
@james-stocks I'm going to wait to release these changes to keep our testing infrastructure stable for in-progress releases.

james-stocks commented:
@kevpl thanks - I'll tidy up the commits now. I'll leave the DO NOT MERGE label until the current releases are out of the way?

pcarlisle commented:
Should this be separate from the disable_updates functionality that's already present?

https://github.com/puppetlabs/beaker/blob/fdd2c7e31ace0d896d334dcbfebce35eb7061952/lib/beaker/options/presets.rb#L164
https://github.com/puppetlabs/beaker/blob/fdd2c7e31ace0d896d334dcbfebce35eb7061952/lib/beaker/host_prebuilt_steps.rb#L389-L396
https://github.com/puppetlabs/beaker/blob/fdd2c7e31ace0d896d334dcbfebce35eb7061952/lib/beaker/hypervisor.rb#L123-L125

james-stocks commented:
@pcarlisle I was hoping that I could simply follow the disable_updates change; but since analytics is PE-only, this change belongs here in beaker-pe.
cc @kevpl

pcarlisle commented:
cool makes sense

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/129/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/131/

kevpl commented:
the 'do not merge' label was on for the releases that were occurring. Those are over now, so we're ready to merge this.
